src/D2L.Bmx/HtmlXmlHelper.cs

```diff
@@ -1,17 +1,23 @@
-using System.Text.RegularExpressions;
+using System.Net;
 using System.Xml;
 using D2L.Bmx.Aws;
+using HtmlAgilityPack;
 
 namespace D2L.Bmx;
 
 internal static partial class HtmlXmlHelper {
 	public static string GetSamlResponseFromLoginPage( string html ) {
-		// TODO: use proper HTML parsing instead of regex
-		var inputXml = new XmlDocument();
-		inputXml.LoadXml( SamlResponseInputRegex().Match( html ).Value );
+		var htmlDoc = new HtmlDocument();
+		htmlDoc.LoadHtml( html );
 
-		return inputXml.SelectSingleNode( "//@value" )?.InnerText
-			?? throw new BmxException( "Error parsing AWS login page" );
+		if(
+			// `SelectSingleNode` can return null even though its signature doesn't say so
+			htmlDoc.DocumentNode.SelectSingleNode( "//input[@name='SAMLResponse']" ) is HtmlNode inputNode
+			&& inputNode.GetAttributeValue( "value", /* default: */ null ) is string samlResponse
```

Comment:
I would've used named arguments, except that this name is `def`, which is confusing as hell...
In my brain "def" == "definition".